src/shared-components/alert/test.tsx

```diff
@@ -14,97 +15,81 @@ const CustomContentComponent = () => (
   </div>
 );
 
-const createNodeMock = (element: ReactElement) => {
-  if (element.type === 'div') {
-    return {
-      innerHTML: alertText,
-    };
-  }
-  return null;
-};
-
 describe('Alert UI snapshots', () => {
   test('renders a default alert', () => {
-    const component = renderer.create(<Alert content={alertText} />, {
-      createNodeMock,
-    });
+    const { container } = render(<Alert content={alertText} />);
 
-    const tree = component.toJSON();
-    expect(tree).toMatchSnapshot();
+    expect(container.firstElementChild).toMatchSnapshot();
   });
 
   test('renders success alert', () => {
-    const component = renderer.create(
-      <Alert content={alertText} type="success" />,
-      { createNodeMock },
-    );
+    const { container } = render(<Alert content={alertText} type="success" />);
 
-    const tree = component.toJSON();
-    expect(tree).toMatchSnapshot();
+    expect(container.firstElementChild).toMatchSnapshot();
   });
 
   test('renders error alert', () => {
-    const component = renderer.create(
-      <Alert content={alertText} type="error" />,
-      { createNodeMock },
-    );
+    const { container } = render(<Alert content={alertText} type="error" />);
 
-    const tree = component.toJSON();
-    expect(tree).toMatchSnapshot();
+    expect(container.firstElementChild).toMatchSnapshot();
   });
 
   test('renders a sticky alert', () => {
-    const component = renderer.create(
+    const { container } = render(
       <Alert content={alertText} duration="sticky" />,
-      { createNodeMock },
     );
 
-    const tree = component.toJSON();
-    expect(tree).toMatchSnapshot();
+    expect(container.firstElementChild).toMatchSnapshot();
   });
 
   test('renders custom component passed in content prop', () => {
-    const component = renderer.create(
+    const { container } = render(
       <Alert content={<CustomContentComponent />} />,
-      { createNodeMock },
     );
 
-    const tree = component.toJSON();
-    expect(tree).toMatchSnapshot();
+    expect(container.firstElementChild).toMatchSnapshot();
   });
 
   test('Alert onExit is triggered on click', () => {
     jest.useFakeTimers();
     const spy = jest.fn();
-    const component = renderer.create(
+    const { container } = render(
       <Alert
         content={<CustomContentComponent />}
         onExit={spy}
         duration="sticky"
       />,
-      { createNodeMock },
     );
 
-    component.root.findByType('button').props.onClick();
-    jest.runAllTimers();
+    assert(container.firstElementChild);
```

Comment:
The DOM type for `firstElementChild` is `Element | null`, so type consistency for `userEvent.click` requires it be non-null. `assert` is a helpful utility added in this PR that automatically handles the type-narrowing + throws an error if our actual code does not our typing of it.